syntax = "proto3";
package kodec;

enum MsgType {
    DEFAULT = 0;
    JOIN_ROOM_REQ = 10;
    JOIN_ROOM_RES = 11;
    USER_LIST_CHANGE_BROADCAST = 12;

    RE_JOIN_REQ = 20;
    RE_JOIN_RES = 21;

    LEAVE_ROOM_REQ = 40;
    LEAVE_ROOM_RES = 41;

    GET_LIVE_INFO_REQ = 50;
    GET_LIVE_INFO_RES = 51;

    GET_PPT_INFO_REQ = 60;
    GET_PPT_INFO_RES = 61;

    GET_PENCIL_LIST_REQ = 70;
    GET_PENCIL_LIST_RES = 71;

    GET_USER_LIST_REQ = 80;
    GET_USER_LIST_RES = 81;

    CHAT_REQ = 90;
    CHAT_RES = 91;
    CHAT_BROADCAST = 92;

    FORBID_CHAT_REQ = 100;
    FORBID_CHAT_RES = 101;
    FORBID_CHAT_BROADCAST = 102;

    LIVE_STATUS_CHANGE_REQ = 110;
    LIVE_STATUS_CHANGE_RES = 111;
    LIVE_STATUS_CHANGE_BROADCAST = 112;

    CHAT_CONTROL_REQ = 120;
    CHAT_CONTROL_RES = 121;
    CHAT_CONTROL_BROADCAST = 122;

    CHANGE_PPT_PAGE_REQ = 130;
    CHANGE_PPT_PAGE_RES = 131;
    CHANGE_PPT_PAGE_BROADCAST = 132;


    PENCIL_REQ = 150;
    PENCIL_RES = 151;
    PENCIL_BROADCAST = 152;

    STREAM_STATUS_CHANGE_REQ = 160;
    STREAM_STATUS_CHANGE_RES = 161;
    STREAM_STATUS_CHANGE_BROADCAST = 162;

    OPEN_CONTEST_REQ = 170;
    OPEN_CONTEST_RES = 171;
    OPEN_CONTEST_BROADCAST = 172;

    SUBMIT_CONTEST_REQ = 180;
    SUBMIT_CONTEST_RES = 181;
    SUBMIT_CONTEST_BROADCAST = 182;

    CLOSE_CONTEST_REQ = 190;
    CLOSE_CONTEST_RES = 191;
    CLOSE_CONTEST_BROADCAST = 192;

    QUERY_CONTEST_REQ = 200;
    QUERY_CONTEST_RES = 201;

    COUNT_DOWN_START_REQ = 210;
    COUNT_DOWN_START_RES = 211;
    COUNT_DOWN_START_BROADCAST = 212;

    COUNT_DOWN_END_REQ = 220;
    COUNT_DOWN_END_RES = 221;
    COUNT_DOWN_END_BROADCAST = 222;

    VOTE_START_REQ = 230;
    VOTE_START_RES = 231;
    VOTE_START_BROADCAST = 232;

    SUBMIT_VOTE_REQ = 240;
    SUBMIT_VOTE_RES = 241;
    SUBMIT_VOTE_BROADCAST = 242;

    VOTE_STOP_REQ = 250;
    VOTE_STOP_RES = 251;
    VOTE_STOP_BROADCAST = 252;

    VOTE_QUERY_REQ = 260;
    VOTE_QUERY_RES = 261;

    GET_FORBID_LIST_REQ = 270;
    GET_FORBID_LIST_RES = 271;

    STAGE_UP_REQ = 280;
    STAGE_UP_RES = 281;
    STAGE_UP_BROADCAST = 282;
    STAGE_UP_P2P = 283;

    STAGE_DOWN_REQ = 290;
    STAGE_DOWN_RES = 291;
    STAGE_DOWN_BROADCAST = 292;
    STAGE_DOWN_P2P = 293;

    WINDOW_MOVE_REQ = 300;
    WINDOW_MOVE_RES = 301;
    WINDOW_MOVE_BROADCAST = 302;

    GET_STAGE_USER_LIST_REQ = 310;
    GET_STAGE_USER_LIST_RES = 311;
    GET_STAGE_USER_LIST_BROADCAST = 312;

    REWARD_INDIVIDUAL_REQ = 320;
    REWARD_INDIVIDUAL_RES = 321;
    REWARD_INDIVIDUAL_BROADCAST = 322;
    REWARD_INDIVIDUAL_P2P = 323;

    REWARD_EVERYONE_REQ = 330;
    REWARD_EVERYONE_RES = 331;
    REWARD_EVERYONE_BROADCAST = 332;

    REWARD_REPORT_REQ = 340;
    REWARD_REPORT_RES = 341;
    REWARD_REPORT_BROADCAST = 342;

    REWARD_QUERY_REQ = 350;
    REWARD_QUERY_RES = 351;

    GRANT_PRIVILEGE_REQ = 360;
    GRANT_PRIVILEGE_RES = 361;
    GRANT_PRIVILEGE_BROADCAST = 362;
    GRANT_PRIVILEGE_P2P = 363;

    REVOKE_PRIVILEGE_REQ = 370;
    REVOKE_PRIVILEGE_RES = 371;
    REVOKE_PRIVILEGE_BROADCAST = 372;
    REVOKE_PRIVILEGE_P2P = 373;

    STAGE_NOTIFICATION_REQ = 380;
    STAGE_NOTIFICATION_RES = 381;
    STAGE_NOTIFICATION_BROADCAST = 382;

    STUDENT_CHAT_REQ = 390;
    STUDENT_CHAT_RES = 391;
    STUDENT_CHAT_BROADCAST = 392;

    GET_CHAT_RECORD_REQ = 400;
    GET_CHAT_RECORD_RES = 401;

    GET_PRIVILEGE_USER_REQ = 410;
    GET_PRIVILEGE_USER_RES = 411;

    REWARD_QUERY_ALL_REQ = 420;
    REWARD_QUERY_ALL_RES = 421;

    REFUSE_STAGE_UP_REQ = 430;
    REFUSE_STAGE_UP_RES = 431;
    REFUSE_STAGE_UP_P2P = 432;

    ADD_VIDEO_LIST_REQ = 440;
    ADD_VIDEO_LIST_RES = 441;
    ADD_VIDEO_LIST_BROADCAST = 442;

    CHANGE_VIDEO_LIST_REQ = 450;
    CHANGE_VIDEO_LIST_RES = 451;
    CHANGE_VIDEO_LIST_BROADCAST = 452;

    GET_FORBID_LIST_WITH_INFO_REQ = 460;
    GET_FORBID_LIST_WITH_INFO_RES = 461;

    STAGE_UP_1V6_REQ = 470;
    STAGE_UP_1V6_RES = 471;
    STAGE_UP_1V6_BROADCAST = 472;

    STAGE_DOWN_1V6_REQ = 480;
    STAGE_DOWN_1V6_RES = 481;
    STAGE_DOWN_1V6_BROADCASET = 482;

    STAGE_MOVE_1V6_REQ = 490;
    STAGE_MOVE_1V6_RES = 491;
    STAGE_MOVE_1V6_BROADCASET = 492;

    GET_STAGE_USER_LIST_1V6_REQ = 500;
    GET_STAGE_USER_LIST_1V6_RES = 501;
    GET_STAGE_USER_LIST_1V6_BROADCAST = 502;

    REWARD_RANK_SHOW_REQ = 510;
    REWARD_RANK_SHOW_RES = 511;
    REWARD_RANK_SHOW_BROADCAST = 512;

    HANDS_UP_REQ = 520;
    HANDS_UP_RES = 521;
    HANDS_UP_BROADCAST = 522;

    FORBID_MIC_REQ = 530;
    FORBID_MIC_RES = 531;
    FORBID_MIC_BROADCAST = 532;

    REWARD_RANK_REFRESH_BROADCAST = 533;

    FORBID_MIC_ALL_REQ = 540;
    FORBID_MIC_ALL_RES = 541;
    FORBID_MIC_ALL_BROADCAST = 542;

    VOTE_QUERY_1V6_REQ = 550;
    VOTE_QUERY_1V6_RES = 551;

    ADD_BOARD_PENCIL_REQ = 560;
    ADD_BOARD_PENCIL_RES = 561;
    ADD_BOARD_PENCIL_BORADCAST = 562;

    CHANGE_BOARD_REQ = 570;
    CHANGE_BOARD_RES = 571;
    CHANGE_BOARD_BROADCAST = 572;

    GET_BOARD_PENCIL_LIST_REQ = 580;
    GET_BOARD_PENCIL_LIST_RES = 581;

    CHANGE_MEDIA_STATUE_REQ = 590;
    CHANGE_MEDIA_STATUE_RES = 591;
    CHANGE_MEDIA_STATUE_BROADCAST = 592;

    QUERY_MEDIA_INFO_REQ = 600;
    QUERY_MEDIA_INFO_RES = 601;

    GET_TEACHER_ONLINE_LIST_REQ = 610;
    GET_TEACHER_ONLINE_LIST_RES = 611;

    MEDIA_LOCATION_CHANGE_REQ = 620;
    MEDIA_LOCATION_CHANGE_RES = 621;
    MEDIA_LOCATION_CHANGE_BROADCAST = 622;

    GET_LIVE_CONFIG_REQ = 630;
    GET_LIVE_CONFIG_RES = 631;

    GET_MEDIA_STATUS_LOCATION_REQ = 640;
    GET_MEDIA_STATUS_LOCATION_RES = 641;

    SET_MEDIA_STATUS_REQ = 650;
    SET_MEDIA_STATUS_RES = 651;
    SET_MEDIA_STATUS_BROADCAST = 652;

    NOTICE_PUBLISH_REQ = 660;
    NOTICE_PUBLISH_RES = 661;
    NOTICE_PUBLISH_BROADCAST = 662;

    NOTICE_DELETE_REQ = 670;
    NOTICE_DELETE_RES = 671;
    NOTICE_DELETE_BROADCAST = 672;

    NOTICE_QUERY_REQ = 680;
    NOTICE_QUERY_RES = 681;
    NOTICE_QUERY_BROADCAST = 682;

    COME_TO_CLASS_REWARD_REQ = 690;
    COME_TO_CLASS_REWARD_RES = 691;
    COME_TO_CLASS_REWARD_BROADCAST = 692;
    COME_TO_CLASS_REWARD_P2P = 693;

    AVAILABLE_REWARD_COUNT_REQ = 700;
    AVAILABLE_REWARD_COUNT_RES = 701;
    AVAILABLE_REWARD_COUNT_BROADCAST = 702;
    AVAILABLE_REWARD_COUNT_P2P = 703;

    VOICE_READ_START_REQ = 710;
    VOICE_READ_START_RES = 711;
    VOICE_READ_START_BROADCAST = 712;

    VOICE_READ_REPORT_REQ = 720;
    VOICE_READ_REPORT_RES = 721;
    VOICE_READ_REPORT_BROADCAST = 722;

    VOICE_READ_QUERY_REQ = 730;
    VOICE_READ_QUERY_RES = 731;
    VOICE_READ_QUERY_BROADCAST = 732;

    VOICE_READ_STOP_REQ = 740;
    VOICE_READ_STOP_RES = 741;
    VOICE_READ_STOP_BROADCAST = 742;

    NEW_SUBMIT_VOTE_BROADCAST = 750;
    VOTE_START_NEW_BROADCAST = 751;

    VOTE_GROUP_COMPETE_QUERY_REQ = 760;
    VOTE_GROUP_COMPETE_QUERY_RES = 761;

    VOTE_NO_FINISH_P2P = 770;
    VOICE_NO_FINISH_P2P = 771;
    SENTENCE_NO_FINISH_P2P = 772;

    JOIN_RTC_ROOM_REQ = 780;
    JOIN_RTC_ROOM_RES = 781;
    JOIN_RTC_ROOM_BROADCAST = 782;

    CHANGE_RTC_ROOM_WINDOW_LIST_REQ = 790;
    CHANGE_RTC_ROOM_WINDOW_LIST_RES = 791;
    CHANGE_RTC_ROOM_WINDOW_LIST_BROADCAST = 792;

    RTC_ROOM_WINDOW_STAND_OUT_REQ = 800;
    RTC_ROOM_WINDOW_STAND_OUT_RES = 801;
    RTC_ROOM_WINDOW_STAND_OUT_BROADCAST = 802;

    RTC_ROOM_WINDOW_RECOVER_REQ = 810;
    RTC_ROOM_WINDOW_RECOVER_RES = 811;
    RTC_ROOM_WINDOW_RECOVER_BROADCASET = 812;

    STAND_OUT_WINDOW_MOVE_REQ = 820;
    STAND_OUT_WINDOW_MOVE_RES = 821;
    STAND_OUT_WINDOW_MOVE_BROADCASET = 822;

    GET_STAND_OUT_WINDOW_LIST_REQ = 830;
    GET_STAND_OUT_WINDOW_LIST_RES = 831;

    FORBID_RTC_MIC_REQ = 840;
    FORBID_RTC_MIC_RES = 841;
    FORBID_RTC_MIC_BROADCAST = 842;

    RTC_HANDS_UP_REQ = 850;
    RTC_HANDS_UP_RES = 851;
    RTC_HANDS_UP_BROADCAST = 852;

    FORBID_RTC_VIDEO_REQ = 860;
    FORBID_RTC_VIDEO_RES = 861;
    FORBID_RTC_VIDEO_BROADCAST = 862;

    READ_SENTENCE_START_REQ = 870;
    READ_SENTENCE_START_RES = 871;
    READ_SENTENCE_START_BROADCAST = 872;

    READ_SENTENCE_SUBMIT_REQ = 880;
    READ_SENTENCE_SUBMIT_RES = 881;

    READ_SENTENCE_QUERY_REQ = 890;
    READ_SENTENCE_QUERY_RES = 891;

    READ_SENTENCE_STOP_REQ = 900;
    READ_SENTENCE_STOP_RES = 901;
    READ_SENTENCE_STOP_BROADCAST = 902;

    VOTE_RANK_GROUP_REQ = 910;
    VOTE_RANK_GROUP_RES = 911;

    QUESTION_SCREENSHOT_SAVE_REQ = 920;
    QUESTION_SCREENSHOT_SAVE_RES = 921;

    FORBID_CHAT_NO_FEEL_REQ = 930;
    FORBID_CHAT_NO_FEEL_RES = 931;
    FORBID_CHAT_NO_FEEL_P2P = 932;
    FORBID_CHAT_NO_FEEL_BROADCAST = 933;
    FORBID_CHAT_NO_FEEL_NOTICE_BROADCAST = 934;

    GET_FORBID_CHAT_NO_FEEL_USER_LIST_REQ = 940;
    GET_FORBID_CHAT_NO_FEEL_USER_LIST_RES = 941;

    GET_FORBID_CHAT_TIME_REQ = 950;
    GET_FORBID_CHAT_TIME_RES = 951;

    GET_TENCENT_CONFIG_REQ = 960;
    GET_TENCENT_CONFIG_RES = 961;

    GET_REMAIN_LOTTERY_COUNT_REQ = 970;
    GET_REMAIN_LOTTERY_COUNT_RES = 971;

    LOTTERY_REQ = 980;
    LOTTERY_RES = 981;

    RTMP_PUSHER_OPERATION_REQ = 990;
    RTMP_PUSHER_OPERATION_RES = 991;
    RTMP_PUSHER_OPERATION_BROADCAST = 992;

    QUESTIONNAIRE_QUERY_REQ = 1000;
    QUESTIONNAIRE_QUERY_RES = 1001;

    QUESTIONNAIRE_REQ = 1010;
    QUESTIONNAIRE_RES = 1011;
    QUESTIONNAIRE_BROADCAST  = 1012;

    QUESTIONNAIRE_ANSWER_REQ = 1020;
    QUESTIONNAIRE_ANSWER_RES = 1021;

    ASISTANT_QUERY_GROUP_QUESTION_RESULT_REQ = 1030;
    ASISTANT_QUERY_GROUP_QUESTION_RESULT_RES = 1031;

    PUBLIC_CLASS_START_STAGE_UP_REQ = 1040;
    PUBLIC_CLASS_START_STAGE_UP_RES = 1041;
    PUBLIC_CLASS_START_STAGE_UP_BROADCAST = 1042;

    PUBLIC_CLASS_HANDS_UP_REQ = 1050;
    PUBLIC_CLASS_HANDS_UP_RES = 1051;

    PUBLIC_CLASS_QUERY_HANDS_UP_USER_LIST_REQ = 1060;
    PUBLIC_CLASS_QUERY_HANDS_UP_USER_LIST_RES = 1061;

    PUBLIC_CLASS_PICK_HANDS_UP_USER_TO_STAGE_REQ = 1070;
    PUBLIC_CLASS_PICK_HANDS_UP_USER_TO_STAGE_RES = 1071;
    PUBLIC_CLASS_PICK_HANDS_UP_USER_TO_STAGE_P2P = 1072;

    PUBLIC_CLASS_USER_STAGE_UP_REQ = 1080;
    PUBLIC_CLASS_USER_STAGE_UP_RES = 1081;
    PUBLIC_CLASS_USER_STAGE_UP_BROADCAST = 1082;
    PUBLIC_CLASS_STAGE_USER_OFF_LINE_P2P = 1083;

    PUBLIC_CLASS_LET_USER_STAGE_DOWN_REQ = 1090;
    PUBLIC_CLASS_LET_USER_STAGE_DOWN_RES = 1091;
    PUBLIC_CLASS_LET_USER_STAGE_DOWN_BROADCAST = 1092;

    PUBLIC_CLASS_STOP_STAGE_UP_REQ = 1100;
    PUBLIC_CLASS_STOP_STAGE_UP_RES = 1101;
    PUBLIC_CLASS_STOP_STAGE_UP_BROADCAST = 1102;

    GET_AVAILABLE_SPECIAL_AWARD_LIST_REQ = 1110;
    GET_AVAILABLE_SPECIAL_AWARD_LIST_RES = 1111;

    SEND_SPECIAL_AWARD_REQ = 1120;
    SEND_SPECIAL_AWARD_RES = 1121;
    SEND_SPECIAL_AWARD_BROADCAST = 1122;

    SPECIAL_AWARD_REPORT_REQ = 1130;
    SPECIAL_AWARD_REPORT_RES = 1131;

    OFF_LINE_USER_LIST_REQ =  1140;
    OFF_LINE_USER_LIST = 1141;

    READY_TO_START_CLASS_REQ = 1150;
    READY_TO_START_CLASS_RES = 1151;
    READY_TO_START_CLASS_BROADCAST = 1152;

    ASISTANT_STAGE_UP_REQ = 1160;
    ASISTANT_STAGE_UP_RES = 1161;
    ASISTANT_STAGE_UP_BROADCAST = 1162;

    ASISTANT_STAGE_DOWN_REQ = 1170;
    ASISTANT_STAGE_DOWN_RES = 1171;
    ASISTANT_STAGE_DOWN_BROADCAST = 1172;

    QUERY_ASISTANT_STAGE_STATUS_REQ = 1180;
    QUERY_ASISTANT_STAGE_STATUS_RES = 1181;

    LOTTERY_BROADCAST = 1192;

    SYS_PUSH_COMMODITY_P2P = 1202;

    MEDIA_DEVICE_SET_REQ = 1210;
    MEDIA_DEVICE_SET_RES = 1211;

    QUERY_MEDIA_DEVICE_STATUS_REQ = 1220;
    QUERY_MEDIA_DEVICE_STATUS_RES = 1221;

    PRIVATE_MSG_REQ = 1230;
    PRIVATE_MSG_RES = 1231;
    PRIVATE_MSG_P2P = 1232;

    GET_PRIVATE_MSG_UIDS_REQ = 1240;
    GET_PRIVATE_MSG_UIDS_RES = 1241;

    GET_PRIVATE_MSG_BY_UID_REQ = 1250;
    GET_PRIVATE_MSG_BY_UID_RES = 1251;

    REPORT_LAST_READ_CHAT_ID_REQ = 1260;
    REPORT_LAST_READ_CHAT_ID_RES = 1261;
    REPORT_LAST_READ_CHAT_ID_P2P = 1262;

    PPT_PENCIL_UNDO_REQ = 1270;
    PPT_PENCIL_UNDO_RES  = 1271;
    PPT_PENCIL_UNDO_BROADCAST  = 1272;

    PPT_PENCIL_REDO_REQ = 1280;
    PPT_PENCIL_REDO_RES  = 1281;
    PPT_PENCIL_REDO_BROADCAST  = 1282;

    BOARD_PENCIL_UNDO_REQ = 1290;
    BOARD_PENCIL_UNDO_RES  = 1291;
    BOARD_PENCIL_UNDO_BROADCAST  = 1292;

    BOARD_PENCIL_REDO_REQ = 1300;
    BOARD_PENCIL_REDO_RES  = 1301;
    BOARD_PENCIL_REDO_BROADCAST  = 1302;

    TEACHER_SCREEN_TO_ASSISTANT_REQ  = 1310;
    TEACHER_SCREEN_TO_ASSISTANT_RES = 1311;

    GET_PPT_ANIMATION_CURRENT_STEP_REQ = 1320;
    GET_PPT_ANIMATION_CURRENT_STEP_RES = 1321;

    GET_EXAMINATION_CONFIG_REQ = 1330;
    GET_EXAMINATION_CONFIG_RES = 1331;

    START_EXAMINATION_REQ = 1340;
    START_EXAMINATION_RES = 1341;
    START_EXAMINATION_BROADCAST = 1342;

    GET_EXAMINATION_RESULT_OVERVIEW_REQ = 1350;
    GET_EXAMINATION_RESULT_OVERVIEW_RES = 1351;

    GET_QUESTION_ANALYSIS_REQ = 1360;
    GET_QUESTION_ANALYSIS_RES = 1361;

    GET_EXAMINATION_STUDENT_RANK_REQ = 1370;
    GET_EXAMINATION_STUDENT_RANK_RES = 1371;

    GET_STUDENT_EXAMINATION_RESULT_REQ = 1380;
    GET_STUDENT_EXAMINATION_RESULT_RES = 1381;

    GET_EXAMINATION_GROUP_RANK_REQ = 1390;
    GET_EXAMINATION_GROUP_RANK_RES = 1391;

    STOP_EXAMINATION_REQ = 1400;
    STOP_EXAMINATION_RES = 1401;
    STOP_EXAMINATION_BROADCAST = 1402;

    GET_COMPLETED_EXAMINATION_LIST_REQ = 1410;
    GET_COMPLETED_EXAMINATION_LIST_RES = 1411;

    REPORT_EXAMINATION_RESULT_REQ = 1420;
    REPORT_EXAMINATION_RESULT_RES = 1421;

    STUDENT_COMPLETE_EXAMINATION_REQ = 1430;
    STUDENT_COMPLETE_EXAMINATION_RES = 1431;

    QUERY_PRAISE_STUDENT_LIST_REQ = 1440;
    QUERY_PRAISE_STUDENT_LIST_RES  = 1441;

    STUDENT_REPORT_LEAVE_LIVE_REQ = 1450;
    STUDENT_REPORT_LEAVE_LIVE_RES = 1451;

    STUDENT_REPORT_BACK_LIVE_REQ = 1460;
    STUDENT_REPORT_BACK_LIVE_RES = 1461;

    CHECK_EXAMINATION_STATUS_REQ = 1470;
    CHECK_EXAMINATION_STATUS_RES = 1471;

    GET_EXAMINATION_BY_ID_REQ = 1480;
    GET_EXAMINATION_BY_ID_RES = 1481;
    GET_LIVE_NOBOOK_EXPERIMENTATION_LIST_REQ = 1490;
    GET_LIVE_NOBOOK_EXPERIMENTATION_LIST_RES = 1491;

    PUSH_LIVE_NOBOOK_EXPERIMENTATION_REQ  = 1500;
    PUSH_LIVE_NOBOOK_EXPERIMENTATION_RES  = 1501;
    PUSH_LIVE_NOBOOK_EXPERIMENTATION_BROADCAST = 1502;

    STOP_LIVE_NOBOOK_EXPERIMENTATION_REQ = 1510;
    STOP_LIVE_NOBOOK_EXPERIMENTATION_RES  = 1511;
    STOP_LIVE_NOBOOK_EXPERIMENTATION_BROADCAST = 1512;

    COMMIT_LIVE_NOBOOK_EXPERIMENTATION_REQ = 1520;
    COMMIT_LIVE_NOBOOK_EXPERIMENTATION_RES = 1521;

    QUERY_NOBOOK_EXPERIMENTATION_GROUP_RANK_REQ = 1530;
    QUERY_NOBOOK_EXPERIMENTATION_GROUP_RANK_RES = 1531;

    QUERY_NOBOOK_EXPERIMENTATION_STUDENT_RANK_REQ = 1540;
    QUERY_NOBOOK_EXPERIMENTATION_STUDENT_RANK_RES = 1541;

    CHECK_NOBOOK_EXPERIMENTATION_STATUS_REQ = 1550;
    CHECK_NOBOOK_EXPERIMENTATION_STATUS_RES = 1551;

    SUBJECTIVE_PHONETIC_START_REQ = 1570;
    SUBJECTIVE_PHONETIC_START_RES = 1571;
    SUBJECTIVE_PHONETIC_START_BROADCAST = 1572;

    SUBJECTIVE_PHONETIC_STOP_REQ = 1580;
    SUBJECTIVE_PHONETIC_STOP_RES = 1581;
    SUBJECTIVE_PHONETIC_STOP_BROADCAST = 1582;

    SUBJECTIVE_PHONETIC_SUBMIT_REQ = 1590;
    SUBJECTIVE_PHONETIC_SUBMIT_RES = 1591;

    STUDENT_QUERY_OPEN_MOUTH_INFO_REQ = 1600;
    STUDENT_QUERY_OPEN_MOUTH_INFO_RES = 1601;

    TEACHER_QUERY_SUBJECTIVE_PHONETIC_REQ = 1610;
    TEACHER_QUERY_SUBJECTIVE_PHONETIC_RES = 1611;

    SUBJECTIVE_PHONETIC_NO_FINISH_P2P = 1620;

    COURSEWARE_CONFIG_REQ = 1630;
    COURSEWARE_CONFIG_RES = 1631;

    COURSEWARE_EXERCISE_START_REQ = 1640;
    COURSEWARE_EXERCISE_START_RES = 1641;
    COURSEWARE_EXERCISE_START_BROADCAST = 1632;

    COURSEWARE_QUESTION_REPORT_REQ = 1650;
    COURSEWARE_QUESTION_REPORT_RES = 1651;

    COURSEWARE_EXERCISE_STOP_REQ = 1660;
    COURSEWARE_EXERCISE_STOP_RES = 1661;
    COURSEWARE_EXERCISE_STOP_BROADCAST = 1662;

    GET_COURSEWARE_EXERCISE_ANALYSIS_REQ = 1670;
    GET_COURSEWARE_EXERCISE_ANALYSIS_RES = 1671;

    GET_COURSEWARE_EXERCISE_STATUS_REQ = 1680;
    GET_COURSEWARE_EXERCISE_STATUS_RES = 1681;

    CHECK_COURSEWARE_EXERCISE_STATUS_REQ = 1690;
    CHECK_COURSEWARE_EXERCISE_STATUS_RES = 1691;

    GET_COURSEWARE_EXERCISE_STUDENT_RANK_REQ = 1700;
    GET_COURSEWARE_EXERCISE_STUDENT_RANK_RES = 1701;

    GET_COURSEWARE_EXERCISE_GROUP_RANK_REQ = 1710;
    GET_COURSEWARE_EXERCISE_GROUP_RANK_RES = 1711;

    GET_GROUP_COMPETITION_CONFIG_REQ = 1720;
    GET_GROUP_COMPETITION_CONFIG_RES = 1721;

    GROUP_COMPETITION_START_REQ = 1730;
    GROUP_COMPETITION_START_RES = 1731;
    GROUP_COMPETITION_START_BROADCAST = 1732;

    GET_GROUP_COMPETITION_VICTORY_STATUS_REQ = 1740;
    GET_GROUP_COMPETITION_VICTORY_STATUS_RES = 1741;

    GET_GROUP_COMPETITION_RIGHT_RATE_RANK_REQ = 1750;
    GET_GROUP_COMPETITION_RIGHT_RATE_RANK_RES = 1751;

    GROUP_COMPETITION_SHOW_VICTORY_STATUS_REQ = 1760;
    GROUP_COMPETITION_SHOW_VICTORY_STATUS_RES = 1761;
    GROUP_COMPETITION_SHOW_VICTORY_STATUS_BROADCAST = 1762;

    GROUP_COMPETITION_PRAISE_STUDENT_REQ = 1770;
    GROUP_COMPETITION_PRAISE_STUDENT_RES = 1771;
    GROUP_COMPETITION_PRAISE_STUDENT_BROADCAST = 1772;

    GROUP_COMPETITION_ENCOURAGE_STUDENT_REQ = 1780;
    GROUP_COMPETITION_ENCOURAGE_STUDENT_RES = 1781;
    GROUP_COMPETITION_ENCOURAGE_STUDENT_BROADCAST = 1782;

    GROUP_COMPETITION_REWARD_STUDENT_REQ = 1790;
    GROUP_COMPETITION_REWARD_STUDENT_RES = 1791;
    GROUP_COMPETITION_REWARD_STUDENT_BROADCAST = 1792;

    GROUP_COMPETITION_STOP_REQ = 1800;
    GROUP_COMPETITION_STOP_RES = 1801;
    GROUP_COMPETITION_STOP_BROADCAST = 1802;

    GROUP_COMPETITION_SHOW_TOTAL_RANK_REQ = 1810;
    GROUP_COMPETITION_SHOW_TOTAL_RANK_RES = 1811;
    GROUP_COMPETITION_SHOW_TOTAL_RANK_BROADCAST = 1812;

    QUERY_PRAISE_STUDENT_LIST_BY_ASSIST_REQ  = 1820;
    QUERY_PRAISE_STUDENT_LIST_BY_ASSIST_RES = 1821;

    GET_GROUP_COMPETITION_STUDENT_CONFIG_REQ = 1830;
    GET_GROUP_COMPETITION_STUDENT_CONFIG_RES = 1831;

    GET_VOICE_GROUP_RANK_REQ  = 1840;
    GET_VOICE_GROUP_RANK_RES = 1841;

    GET_READ_GROUP_RANK_REQ = 1850;
    GET_READ_GROUP_RANK_RES = 1851;

    GET_ONLINE_GROUP_LIST_REQ = 1860;
    GET_ONLINE_GROUP_LIST_RES = 1861;

    ENCOURAGE_PRAISE_GROUP_REQ = 1870;
    ENCOURAGE_PRAISE_GROUP_RES = 1871;
    ENCOURAGE_PRAISE_GROUP_BROADCAST = 1872;

    JOIN_ROOM_BROADCAST = 1880;
    
    VOTE_GROUP_RANK_QUERY_REQ = 1890;
    VOTE_GROUP_RANK_QUERY_RES  = 1891;

    CREATE_TAG_REQ = 1900;
    CREATE_TAG_RES = 1901;

    CONTINUE_RIGHT_P2P = 1910;

    MVP_BROADCAST = 1920;
    
    PUSH_USER_AND_GROUP_INFO_P2P = 1930;

    PUSH_PET_STATES_P2P = 1940;

    CLOSE_PRAISE_STUDENT_LIST_REQ = 1950;
    CLOSE_PRAISE_STUDENT_LIST_RES = 1951;
    CLOSE_PRAISE_STUDENT_LIST_BROADCAST = 1952;
    PRAISE_STUDENT_LIST_BROADCAST = 1953;

    SUBJECTIVE_PHONETIC_SUBMIT_NEW_REQ  = 1960;

    ASISTANT_LIVE_END_STAGE_UP_REQ = 1970;
    ASISTANT_LIVE_END_STAGE_UP_RES = 1971;

    ASISTANT_LIVE_END_STAGE_DOWN_REQ = 1980;
    ASISTANT_LIVE_END_STAGE_DOWN_RES = 1981;

    QUERY_PRAISE_STUDENT_LIST_GROUP_BY_GROUP_REQ = 1990;
    QUERY_PRAISE_STUDENT_LIST_GROUP_BY_GROUP_RES = 1991;

    QUERY_ENROLL_STUDENT_LIST_REQ = 2000;
    QUERY_ENROLL_STUDENT_LIST_RES = 2001;

    GET_LIVE_MIDDLE_AD_REQ = 2010;
    GET_LIVE_MIDDLE_AD_RES = 2011;

    LIVE_MIDDLE_AD_SHOW_REQ = 2020;
    LIVE_MIDDLE_AD_SHOW_RES = 2021;
    LIVE_MIDDLE_AD_SHOW_BROADCAST = 2022;

    LIVE_MIDDLE_AD_SHOW_END_REQ = 2030;
    LIVE_MIDDLE_AD_SHOW_END_RES = 2031;
    LIVE_MIDDLE_AD_SHOW_END_BROADCAST = 2032;

    GET_LIVE_MIDDLE_AD_LIST_REQ = 2040;
    GET_LIVE_MIDDLE_AD_LIST_RES = 2041;

    SHOW_LIVE_MIDDLE_AD_ITEM_REQ = 2050;
    SHOW_LIVE_MIDDLE_AD_ITEM_RES = 2051;

    GET_MIXED_COURSEWARE_CONFIG_REQ = 2060;
    GET_MIXED_COURSEWARE_CONFIG_RES = 2061;

    SUPER_SMALL_CLASS_RTC_CHANNEL_INFO_REQ = 2070;
    SUPER_SMALL_CLASS_RTC_CHANNEL_INFO_RES = 2071;

    QUERY_SMALL_CLASS_TEAM_INFO_REQ = 2080;
    QUERY_SMALL_CLASS_TEAM_INFO_RES = 2081;

    GIVE_THE_THUMBS_UP_TO_CLASSMATE_REQ = 2090;
    GIVE_THE_THUMBS_UP_TO_CLASSMATE_RES = 2091;
    GIVE_THE_THUMBS_UP_TO_CLASSMATE_P2P = 2092;

    GIVE_THE_THUMBS_UP_NUMBER_P2P = 2100;

    GEN_RUSH_REWARD_REQ = 2110;
    GEN_RUSH_REWARD_RES = 2111;
    GEN_RUSH_REWARD_BROADCAST = 2112;

    DO_RUSH_REWARD_REQ = 2120;
    DO_RUSH_REWARD_RES = 2121;
    DO_RUSH_REWARD_BROADCAST = 2122;

    GET_REWARD_RANK_LIST_REQ = 2130;
    GET_REWARD_RANK_LIST_RES = 2131;

    GET_STUDENT_RANK_LIST_REQ = 2140;
    GET_STUDENT_RANK_LIST_RES = 2141;

    GET_SMALL_CLASS_ALI_VOICE_TOKEN_REQ = 2150;
    GET_SMALL_CLASS_ALI_VOICE_TOKEN_RES = 2151;

    CHANGE_SMALL_CLASS_STUDENT_VIDEO_STATUS_REQ = 2160;
    CHANGE_SMALL_CLASS_STUDENT_VIDEO_STATUS_RES = 2161;
    CHANGE_SMALL_CLASS_STUDENT_VIDEO_STATUS_BROADCAST = 2162;

    QUERY_SMALL_CLASS_TEAM_LIST_REQ = 2170;
    QUERY_SMALL_CLASS_TEAM_LIST_RES = 2171;

    STOP_RUSH_REWARD_REQ = 2180;
    STOP_RUSH_REWARD_RES = 2181;
    STOP_RUSH_REWARD_BROADCAST = 2182;
    
    RING_THE_BELL_REQ = 2190;
    RING_THE_BELL_RES = 2191;
    RING_THE_BELL_BROADCAST = 2192;

    QUERY_OFFLINE_DOUBLE_TEACHER_RTC_REQ = 2200;
    QUERY_OFFLINE_DOUBLE_TEACHER_RTC_RES = 2201;
    
    RUSH_REWARD_COUNT_REQ = 2210;
    RUSH_REWARD_COUNT_RES = 2211;

    RUSH_REWARD_NO_FINISH_P2P = 2220;

    NOTIFY_STUDENT_REWARD_COUNT_P2P = 2230;

    NOTIFY_STUDENT_VIDEO_VIOLATION_REQ = 2240;
    NOTIFY_STUDENT_VIDEO_VIOLATION_RES = 2241;
    NOTIFY_STUDENT_VIDEO_VIOLATION_P2P = 2242;
}
